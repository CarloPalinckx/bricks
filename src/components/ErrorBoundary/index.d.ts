import { Component, ErrorInfo } from 'react';
declare type PropsType = {
    hasError?: boolean;
    message: string;
    reportError(error: Error, errorInfo: ErrorInfo): void;
};
declare type StateType = {
    hasError: boolean;
};
declare class ErrorBoundary extends Component<PropsType, StateType> {
    constructor(props: PropsType);
    componentDidCatch(error: Error, errorInfo: ErrorInfo): void;
    render(): JSX.Element;
}
export default ErrorBoundary;
export { PropsType, StateType };
