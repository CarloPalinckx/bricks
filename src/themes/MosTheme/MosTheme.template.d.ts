import { StatelessComponent } from 'react';
declare const MosTheme: StatelessComponent;
export default MosTheme;
