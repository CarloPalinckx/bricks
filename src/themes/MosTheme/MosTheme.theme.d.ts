import ThemeType from '../../types/ThemeType';
declare const theme: ThemeType;
export default theme;
