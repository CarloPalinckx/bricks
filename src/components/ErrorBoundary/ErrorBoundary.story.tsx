import { select } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import ErrorBoundary from '.';
import Text from '../Text';

storiesOf('ErrorBoundary', module)
    .add('Default', () => {
        const hasError = select(
            'hasError',
            ['true', 'false'],
            'false'
        );

        return (
            <ErrorBoundary
                message="This is an error"
                hasError={hasError === 'true'}
                reportError={():void => {
                    // report error
                }}
            >
                <Text>This does not have an error.</Text>
            </ErrorBoundary>
        );
    });
