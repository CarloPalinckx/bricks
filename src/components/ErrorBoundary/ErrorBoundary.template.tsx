import React, { Component, ErrorInfo } from 'react';
import Notification from '../Notification';

type PropsType = {
    hasError?:boolean;
    message:string;
    reportError(error:Error, errorInfo:ErrorInfo):void;
};

type StateType = {
    hasError:boolean;
};

class ErrorBoundary extends Component<PropsType, StateType> {
    public constructor(props:PropsType) {
        super(props);

        this.state = {
            hasError: this.props.hasError !== undefined
                ? this.props.hasError
                : false,
        };
    }

    public componentDidCatch(error:Error, errorInfo:ErrorInfo):void {
        this.setState({
            hasError: true,
        });

        this.props.reportError(error, errorInfo);
    }

    public render():JSX.Element {
        return this.state.hasError || (this.props.hasError !== undefined && this.props.hasError)
            ? <Notification severity="error" message={this.props.message} />
            : <div>{this.props.children}</div>;
    }
}

export default ErrorBoundary;
export {
    PropsType,
    StateType
};
