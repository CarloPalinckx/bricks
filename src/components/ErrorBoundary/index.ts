export { default as default } from './ErrorBoundary.template';
