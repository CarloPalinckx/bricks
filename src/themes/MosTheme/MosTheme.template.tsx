import React, { StatelessComponent } from 'react';
import { ThemeProvider } from '../../utility/styled';
import theme from './MosTheme.theme';

const MosTheme: StatelessComponent = (props): JSX.Element => {
    return <ThemeProvider theme={theme}>{props.children}</ThemeProvider>;
};

export default MosTheme;
