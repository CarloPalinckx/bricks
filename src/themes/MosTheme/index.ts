export { default } from './MosTheme.template';
export { default as mosTheme } from './MosTheme.theme';
