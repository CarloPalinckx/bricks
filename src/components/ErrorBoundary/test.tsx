import React from 'react';
import Renderer from 'react-test-renderer';
import ErrorBoundary from '.';
import MosTheme from '../../themes/MosTheme';

/* tslint:disable */
(console.error as any).mockImplementationOnce(() => { });
/* tslint:enable */

describe('ErrorBoundary', () => {
    beforeEach(() => {
        /*
         * this the virtual console of jsdom
         * because jest doesn't swallow jsdom errors in its console and react
         * doesn't swallow caught errors even if they are caught.
         */

        /* tslint:disable */
        (global as any).console = {
            error: jest.fn(),
        };

        jest
            .spyOn((window as any)._virtualConsole, 'emit')
            .mockImplementation(() => false);
        /* tslint:enable */
    });

    it('should catch an error and report it', () => {
        const mockReport = jest.fn();

        const ThrowingComponent = (): JSX.Element => {
            throw new Error('Mock Error');
        };

        const boundary = Renderer.create(
            <MosTheme>
                <ErrorBoundary
                    message="This has an error"
                    reportError={mockReport}
                >
                    <ThrowingComponent />
                </ErrorBoundary>
            </MosTheme>,
        ).toJSON();

        expect(boundary).toMatchSnapshot();
        expect(mockReport).toHaveBeenCalled();
    });
});
