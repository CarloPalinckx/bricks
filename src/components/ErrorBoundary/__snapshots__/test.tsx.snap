// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should catch an error and report it 1`] = `
.c3 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  height: 18px;
  width: 18px;
}

.c3 svg {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  max-height: 100%;
  max-width: 100%;
  fill: currentColor;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 12px 12px 12px 12px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0px 12px 0px 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c0 {
  border-width: 1px;
  border-style: solid;
  font-family: Source Sans Pro,sans-serif;
  font-size: 14px;
  background-color: #ffbdce;
  border-color: #ff6c93;
  color: #c81746;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span
        aria-hidden={true}
        className="c3"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test-file-stub",
          }
        }
        role="img"
      />
    </div>
    This has an error
  </div>
</div>
`;
